dbfit-java/derby/src/main/java/dbfit/environment/DerbyEnvironment.java

```diff
@@ -51,42 +52,192 @@ public Pattern getParameterPattern() {
         return paramRegex;
     }
 
-    public Map<String, DbParameterAccessor> getAllColumns(
-            final String tableOrViewName) throws SQLException {
-        String qry = "SELECT COLUMNNAME, COLUMNDATATYPE "
-                + "FROM SYS.SYSCOLUMNS WHERE REFERENCEID = "
-                + "(SELECT TABLEID FROM SYS.SYSTABLES WHERE TABLENAME = ?)";
-        return readIntoParams(tableOrViewName.toUpperCase(), qry);
+    @Override
+    public Map<String, DbParameterAccessor> getAllColumns(String tableName)
+            throws SQLException {
+        return new DatabaseTable(buildDatabaseObjectName(tableName)).getParams();
     }
 
-    private Map<String, DbParameterAccessor> readIntoParams(
-            String tableOrViewName, String query) throws SQLException {
-        try (PreparedStatement dc = getConnection().prepareStatement(query)) {
-            dc.setString(1, tableOrViewName);
+    private abstract class DatabaseObject {
+        protected DatabaseObjectName name;
+
+        private DatabaseObject(DatabaseObjectName objName) {
+            name = objName;
+        }
+
+        protected DatabaseMetaData getDbMetaData() throws SQLException {
+            return getConnection().getMetaData();
+        }
+
+        abstract protected ResultSet getObjectMetaData() throws SQLException;
+
+        public boolean exists() throws SQLException {
+            try (ResultSet rs = getObjectMetaData()) {
+                return rs.next();
+            }
+        }
+
+        abstract protected ResultSet getColumns() throws SQLException;
 
-            ResultSet rs = dc.executeQuery();
+        public Map<String, DbParameterAccessor> getParams() throws SQLException {
             Map<String, DbParameterAccessor> allParams = new HashMap<String, DbParameterAccessor>();
```

Comment:
Should be fine to call `new HashMap<>()` here, no need to list types on the right side (I think we already did that somewhere).